src/main/java/build/buildfarm/common/grpc/StubWriteOutputStream.java

```diff
@@ -259,7 +259,7 @@ public void write(int b) throws IOException {
   @Override
   public synchronized boolean isReady() {
     if (writeObserver == null) {
-      return false;
+      throw new RuntimeException("isReady should not be called before getOutput");
```

Comment:
Prefer `checkNotNull` with the message here - the ISE that it throws is better than RE.